include/server.h

```diff
@@ -52,7 +52,7 @@ namespace kiwix
        void stop();
 
        void setRoot(const std::string& root);
-       void setAddress(const std::string& addr) { m_addr = addr; }
+       void setAddress(const std::string& addr) { if(!addr.empty()) addr.find('[') != std::string::npos ? m_addr.addr6 = addr : m_addr.addr = addr; }
```

Comment:
1. Move the definition to .cpp and implement as a multiliner using an `if` statement instead of the conditional expression.
2. Is `[` a mandatory part of an IPv6 address syntax? Isn't it better to test for the presence of a colon (`:`)?